### PR TITLE
Add 2.10.0 release notes

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description
Add release notes for 2.10 release

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-notifications.release-notes-2.10.0.0.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## Version 2.10.0.0 2023-09-19
```
Comment (xluo-aws) [inline release-notes/opensearch-notifications.release-notes-2.10.0.0.md:1]: Release date will be 09/25

File: release-notes/opensearch-notifications.release-notes-2.10.0.0.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+### Maintenance
+* [AUTO] Increment version to 2.10.0-SNAPSHOT([#706](https://github.com/opensearch-project/notifications/pull/706))
+
+### Infrastructure
+* Fix core refactor: StreamIO from common to core.common([#707](https://github.com/opensearch-project/notifications/pull/707))
+* Fix core XcontentFactory refactor([#732](https://github.com/opensearch-project/notifications/pull/732))
+* Fix actions components after core([#739](https://github.com/opensearch-project/notifications/pull/739))
+* Add auto release workflow([#731](https://github.com/opensearch-project/notifications/pull/731))
+* Onboarding system and hidden index([#731](https://github.com/opensearch-project/notifications/pull/731))
```
Comment (xluo-aws) [inline release-notes/opensearch-notifications.release-notes-2.10.0.0.md:18]: This should be 748, not 731

File: release-notes/opensearch-notifications.release-notes-2.10.0.0.md (hunk 1 above)
Comment (Hailong-am) [reply release-notes/opensearch-notifications.release-notes-2.10.0.0.md]: update to latest release date

File: release-notes/opensearch-notifications.release-notes-2.10.0.0.md (hunk 2 above)
Comment (Hailong-am) [reply release-notes/opensearch-notifications.release-notes-2.10.0.0.md]: good catch, update to #742

**Review by xluo-aws — APPROVED** [review]

**Review by zhichao-aws — APPROVED** [review]

**Pull request merged**